File: Java/pangram.java
import java.util.Scanner;
import java.util.HashMap;

class Main{
    public static void main(String args[]){
        String sentence = input();
        output(sentence);
    }

    public static String input(){
        Scanner scanner =  new Scanner(System.in);
        return scanner.nextLine();
    }

    public static void output(String sentence){
        System.out.print(check(sentence));
    }

    public static boolean check(String sentence){
        HashMap<Character, Integer> letters = new HashMap<>();
        for(int i = 0; i < sentence.length(); i++){
            letters.put(sentence.charAt(i), 1);
        }
        boolean isPan = true;
        for (char c = 97; c <= 122; c++){
            if (letters.get(c) == null){
            System.out.println("There is no " + c);
            isPan = false;
            }
        }
        return isPan;
    }
}

File: Rust/pangram.rs
use std::collections::HashMap;
use std::io;

fn main() {
    let mut sentence = String::new();
    input(&mut sentence);
    output(&sentence);
}

fn input(sentence: &mut String) {
    let _ = io::stdin().read_line(sentence);
    *sentence = sentence.to_string();
}

fn output(sentence: &str) {
    print!("{}", check(sentence));
}

fn check(sentence: &str) -> bool {
    let mut letters: HashMap<char, i32> = HashMap::new();
    for s in sentence.chars() {
        *letters.entry(s).or_insert(0) += 1;
    }
    
    let mut is_pan = true;
    for c in 'a'..'z' {
        let c = c as char;
        if *letters.get(&c).unwrap_or(&0) == 0 {
            println!("There is no {}", c);
            is_pan = false;
        }
    }
    is_pan
}